sling/nlp/wikicat/app/wikicat.js

```diff
@@ -160,15 +191,98 @@ app.controller('SearchCtrl', function($scope, $http,  $location, $mdDialog) {
   $scope.browseSignature = function(sig) {
     $scope.query = sig;
     $scope.parse_selector = "True";
+    $scope.selected_parses.clear();
     $scope.search();
   }
 
   // Click-handler for navigating to a category. Shows only parses selected
   // by any currently-specified selection formula.
   $scope.browseCategory = function(name) {
     $scope.query = name;
+    $scope.selected_parses.clear();
     $scope.search();
   }
+
+  // Returns whether a particular row index should be displayed or not.
+  $scope.showTableRow = function(row) {
+    var start = $scope.current_page * $scope.page_size;
+    return (start <= row) && (row < start + $scope.page_size);
+  }
+
+  // Returns a [start, end) range list with the given step size.
+  $scope.range = function(start, end, step) {
```

Comment:
JS not having standard libraries is just so awesome...

Reply:
Yes, so annoying...